src/intrinsics/prepack/global.js

```diff
@@ -120,19 +120,29 @@ export default function(realm: Realm): void {
   //     that is not subsequently applied, the function will not be registered
   //     (because prepack won't have a correct value for the FunctionValue itself)
   global.$DefineOwnProperty("__optimize", {
-    value: new NativeFunctionValue(realm, "global.__optimize", "__optimize", 0, (context, [value, config]) => {
-      // only optimize functions for now
-      if (value instanceof ECMAScriptSourceFunctionValue) {
-        realm.assignToGlobal(
-          t.memberExpression(
-            t.memberExpression(t.identifier("global"), t.identifier("__optimizedFunctions")),
-            t.identifier("" + additonalFunctionUid++)
-          ),
-          value
-        );
+    value: new NativeFunctionValue(
+      realm,
+      "global.__optimize",
+      "__optimize",
+      0,
+      (context, [value, argModelString, config]) => {
+        // only optimize functions for now
+        if (value instanceof ECMAScriptSourceFunctionValue) {
+          let functionToOptimize = new ObjectValue(realm, realm.intrinsics.ObjectPrototype);
```

Comment:
It's not ` functionToOptimize` anymore. It's now a `functionDescriptor` or something like that.